ads/google/doubleclick.js

```diff
@@ -265,3 +244,51 @@ function doubleClickWithGlade(global, data, gladeExperiment) {
 function getCorrelator(global) {
   return makeCorrelator(global.context.clientId, global.context.pageViewId);
 }
+
+function centerAd() {
+  setStyles(dev().assertElement(global.document.querySelector('#c')), {
+    top: '50%',
+    left: '50%',
+    bottom: '',
+    right: '',
+    transform: 'translate(-50%, -50%)',
+  });
+}
+
+/**
+ * @param {Object} data
+ * @return {!string}
+ */
+export function selectGptExperiment(data) {
+  const fileExperimentConfig = {
+    21060540: 'gpt_sf_a.js',
+    21060541: 'gpt_sf_b.js',
+  };
+  // Note that reduce will return the first item that matches but it is
+  // expected that only one of the experiment ids will be present.
+  let expFilename;
+  (data['experimentId'] || '').split(',').forEach(
+    val => expFilename = expFilename || fileExperimentConfig[val]);
+  return expFilename;
+}
+
+export function writeAdScript(data, gptFilename) {
+  const url =
+  `https://www.googletagservices.com/tag/js/${gptFilename || 'gpt.js'}`;
+  if (gptFilename || data.useSameDomainRenderingUntilDeprecated != undefined
```

Comment:
Maybe I'm mistaken but I think that the function is called twice but it's passed different arguments (specifically the experiment branch) each time. So I'd still need the branched logic to set the experiment condition even if I reduced the calls to the function to one. What do you think?